tests/e2e/default/client-side-transitions/create-user-for-contacts.wdio-spec.js

```diff
@@ -0,0 +1,402 @@
+const { BASE_URL, DEFAULT_USER_CONTACT_DOC } = require('../../../constants');
+const contactsPage = require('../../../page-objects/default/contacts/contacts.wdio.page');
+const genericForm = require('../../../page-objects/default/enketo/generic-form.wdio.page');
+const sentinelUtils = require('../../../utils/sentinel');
+const replaceUserForm = require('../../../page-objects/default/enketo/replace-user.wdio.page');
+
+const DISTRICT = {
+  _id: 'fixture:district',
+  type: 'district_hospital',
+  contact: {
+    _id: 'fixture:user:original_person',
+  }
+};
+
+const ORIGINAL_USER = {
+  username: `original_person`,
+  password: 'medic.123',
+  place: DISTRICT._id,
+  phone: '+254712345678',
+  contact: { _id: 'fixture:user:original_person', name: 'Original Person', role: 'chw' },
+  roles: ['chw'],
+};
+
+const ONLINE_USER = {
+  username: `online_user`,
+  password: 'medic.123',
+  place: DISTRICT._id,
+  phone: '+254712345678',
+  contact: { _id: 'fixture:user:online_person', name: 'Online Person', role: 'chw' },
+  roles: ['program_officer'],
+};
+
+const newUsers = [];
+
+const getQueuedMessages = () => utils.db.query('medic-admin/message_queue', { reduce: false, include_docs: true })
+  .then(response => response.rows.map(row => row.doc));
+
+const SETTINGS = {
+  transitions: { create_user_for_contacts: true },
+  create_user_for_contacts: { replace_forms: ['replace_user'] },
+  token_login: { enabled: true },
+  app_url: BASE_URL
+};
+
+const BASIC_FORM_DOC = {
+  _id: 'form:basic_form',
+  internalId: 'basic_form',
+  title: 'Form basic_form',
+  type: 'form',
+  _attachments: {
+    xml: {
+      content_type: 'application/octet-stream',
+      data: Buffer.from(fs.readFileSync(`${__dirname}/forms/basic_form.xml`, 'utf8')).toString('base64')
+    }
+  }
+};
+
+const loginAsOfflineUser = async () => {
+  await utils.createUsers([ORIGINAL_USER]);
+  newUsers.push(ORIGINAL_USER.username);
+  await loginPage.login(ORIGINAL_USER);
+  await commonPage.waitForPageLoaded();
+  await commonPage.closeTour();
+  return ORIGINAL_USER.contact._id;
+};
+
+const loginAsOnlineUser = async () => {
+  await utils.createUsers([ONLINE_USER]);
+  newUsers.push(ONLINE_USER.username);
+  await loginPage.login(ONLINE_USER);
+  await commonPage.waitForPageLoaded();
+  await commonPage.closeTour();
+  return ONLINE_USER.contact._id;
+};
+
+const submitReplaceUserForm = async () => {
+  await contactsPage.createNewAction('Replace User');
+  await replaceUserForm.selectAdminCode('secretCode');
+  await genericForm.nextPage();
+  await replaceUserForm.selectContactFullName('Replacement User');
+  await replaceUserForm.selectContactDobUnknown();
+  await replaceUserForm.selectContactAgeYears(22);
+  await replaceUserForm.selectContactSex(replaceUserForm.SEX.female);
+  await genericForm.nextPage();
+  await (await genericForm.submitButton()).click();
+  await commonPage.waitForPageLoaded();
+};
+
+const sync = async () => {
+  await commonPage.openHamburgerMenu();
```

Comment:
`commonPage.sync` [waits for the sync success message to be displayed](https://github.com/medic/cht-core/blob/5ff7749e3b66b3e54edd5ee904f3d13665f7abbb/tests/page-objects/default/common/common.wdio.page.js#L191).  However, this might not happen if the logout functionality runs first.